Mathlib/Data/ZMod/Basic.lean

```diff
@@ -934,6 +934,19 @@ theorem val_eq_zero : ∀ {n : ℕ} (a : ZMod n), a.val = 0 ↔ a = 0
 theorem val_ne_zero {n : ℕ} (a : ZMod n) : a.val ≠ 0 ↔ a ≠ 0 :=
   (val_eq_zero a).not
 
+-- should this be a simp lemma?
+theorem val_eq_one {n : ℕ} [h : n.AtLeastTwo] (a : ZMod n) :
+    a.val = 1 ↔ a = 1 := by
+  rcases n with (⟨⟩|⟨n⟩)
+  · have := h.prop
+    simp at this
+  · rcases n with (⟨⟩|⟨n⟩)
+    · have := h.prop
+      norm_num at this
+    · rw [Fin.ext_iff]
+      simp only [Fin.val_one]
+      exact Iff.rfl
```

Comment:
See https://github.com/YaelDillies/LeanAPAP/blob/master/LeanAPAP/Mathlib/Data/ZMod/Basic.lean#L25-L30 for ideas